RISK_MATRIX.md

```diff
@@ -0,0 +1,61 @@
+# Risk consideration
+
+When merging a new feature of considerable size or modifying an existing one,
+consider adding a *Risk Matrix* section to your PR in collaboration with other
+developers on your team and the QA team.
+
+Below are some general themes to consider for the *Risk Matrix*:
+
+
+## General risks
+
+- What happens when your feature is used in a non-default space or a custom
+  space?
+- What happens when Kibana is run in a cluster mode?
+- What happens when plugin you depend on is disabled?
+- What happens when feature you depend on is disabled?
+- Is your change working correctly regardless of `kibana.yml` configuration or
+  UI Setting configuration? (For example, does it support both
+  `state:storeInSessionStorage` UI setting states?)
+- What happens when a third party integration you depend on is not responding?
+- How is authentication handled with third party services?
+- Does the feature work in Elastic Cloud?
+- Does the feautre create a setting that needs to be exposed, or configured
+  differently than the default, on the Elastic Cloud?
+- Is there a significant performance impact that may affect Cloud Kibana
+  instances?
+- Does your feature need to be aware of running in a container?
+- Does the feature Work with security disabled, or fails gracefully?
```

Comment:
I think this is redundant with: "What happens when a plugin you depend on is disabled"

Reply:
There are two ways to "disable security" at the moment:
1) Disable security in Elasticsearch by setting `xpack.security.enabled: false` in the `elasticsearch.yml`. This is the recommended approach.
2) Disable Kibana's security plugin by setting `xpack.security.enabled: false` in the `kibana.yml`. This will be removed starting in 8.0 as it's super confusing and complicated to support.